### ISSUE TITLE
Length of AES Encrypted output is doubtful

### ISSUE THREAD

**Issue by arihantdaga** [brix/crypto-js#137]
if i encrypt a string of length 112 i should recieve an Encrypted string of lenght 112 right ? and not 128. But i am getting lenght of encrypted string as 128 . 
Same is the issue with decryption. 
Actual problem is that i am using a C library in my microcontroller to get aes encrypted string which gives me encrypted string of length 112 for the string of length112. Which i am not able to decrypt here in nodeJs, it's throwing an error cant (Invalid array length).

I am posting the code snippet i used for encrypting in nodeJs... 

```
function encrypt(message, iv, secretKey){
	iv = CryptoJS.enc.Hex.parse(iv);
	let key= CryptoJS.enc.Hex.parse( secretKey );
	console.log("Let's Encrypt");
	var bytes  = CryptoJS.AES.encrypt( message, key , { iv: iv} );
	console.log(bytes.ciphertext.sigBytes);
	message = bytes.toString(CryptoJS.AES.Base64);
	let finalObj = {meta:CryptoJS.enc.Base64.stringify(iv), payload:message };
	console.log(finalObj);
}

const message = JSON.stringify({"power":{"power":0},"switch":{"relays":[0,0,0],"dimmer":0},"sns":{"temphumid":"0,1","analog":"0","motion":"0"}});

const iv = "5567b85c5226c5d521e528aca146e2e9";

encrypt(message, iv, "44a47d4f393b1497c6154f3046669317");


```

Any help would be appreciated.. Am i doing something wrong ?

**Comment by DanBeard** [comment]
No,  typically in AES, you would expect the ciphertext to always be a multiple of the block size (128) so if the message isn't already a multiple of that size, the last block needs to be padded.

By default  crypto-js uses CBC mode and the PKCS7 padding function to pad messages to a multiple of 128 bytes. It similarly expects, by default, that any ciphertext it gets will be in CBC mode and padded by PKCS7. 

in your example if you change to using no padding like this then you get 112 byte output which might work for your needs , but no guarantees (Check the mode used on the microcontroller too... is it CBC? ):
```javascript
var bytes  = CryptoJS.AES.encrypt( message, key , { iv: iv, padding:CryptoJS.pad.NoPadding } );
console.log(bytes.ciphertext.sigBytes);
```

**Comment by sergio-domingues** [comment]
@DanBeard 128 bits* . Where do I find the specification saying that the default padding is PKCS7? I traversed the Crypto documentation and didnt find any reference..

**Comment by ayushsharma82** [comment]
`console.log(ebytes.ciphertext.sigBytes);`
 works and returns correct value!
Thanks for the help @DanBeard . and also i would like to ask if sending cipher length + the cipher, itself out in wild will pose any security risk?

**Comment by ash-jha** [comment]
@sergio-domingues see the docs in the Ciphers section https://cryptojs.gitbook.io/docs/#the-cipher-input 

"And CryptoJS supports the following padding schemes:
Pkcs7 (the default)
Iso97971
AnsiX923
Iso10126
ZeroPadding
NoPadding
"
